src/parser.yy

```diff
@@ -220,8 +224,8 @@ start:          START_PROGRAM program END { driver.result = $2; }
                 ;
 
 program:
-                c_definitions config imports map_decl_list macros body {
-                    $$ = driver.ctx.make_node<ast::Program>($1, $2, std::move($3), std::move($4), std::move($5), std::move($6.second), std::move($6.first), @$);
+                c_definitions config opts imports map_decl_list macros body {
```

Comment:
We probably need to allow for flexibility in the ordering of these blocks. Or at least grouping them, e.g. (c_definitions|config|opts|imports)*(map_decl_list|macros|body)*